BHoMUpgrader32/Converter.cs

```diff
@@ -37,14 +37,71 @@ public class Converter : Base.Converter
+            ToNewObject.Add("BH.oM.Structure.Results.BarStrain", UpgradeResult);
+            ToNewObject.Add("BH.oM.Structure.Results.BarStress", UpgradeResult);
+            ToNewObject.Add("BH.oM.Structure.Results.CompositeUtilisation", UpgradeResult);
+            ToNewObject.Add("BH.oM.Structure.Results.SteelUtilisation", UpgradeResult);
+            ToNewObject.Add("BH.oM.Structure.Results.GlobalReactions", UpgradeResult);
+            ToNewObject.Add("BH.oM.Structure.Results.ModalDynamics", UpgradeResult);
+            ToNewObject.Add("BH.oM.Structure.Results.StructuralGlobalResult", UpgradeResult);
+            ToNewObject.Add("BH.oM.Structure.Results.MeshDisplacement", UpgradeResult);
+            ToNewObject.Add("BH.oM.Structure.Results.MeshForce", UpgradeResult);
+            ToNewObject.Add("BH.oM.Structure.Results.MeshElementResult", UpgradeResult);
+            ToNewObject.Add("BH.oM.Structure.Results.MeshResult", UpgradeMeshResult);
+            ToNewObject.Add("BH.oM.Structure.Results.MeshStress", UpgradeResult);
+            ToNewObject.Add("BH.oM.Structure.Results.MeshVonMises", UpgradeResult);
+            ToNewObject.Add("BH.oM.Structure.Results.NodeAcceleration", UpgradeResult);
+            ToNewObject.Add("BH.oM.Structure.Results.NodeDisplacement", UpgradeResult);
+            ToNewObject.Add("BH.oM.Structure.Results.NodeReaction", UpgradeResult);
+            ToNewObject.Add("BH.oM.Structure.Results.NodeResult", UpgradeResult);
+            ToNewObject.Add("BH.oM.Structure.Results.NodeVelocity", UpgradeResult);
         }
 
 
         /***************************************************/
         /**** Private Methods                           ****/
         /***************************************************/
 
+        public static Dictionary<string, object> UpgradeResult(Dictionary<string, object> result)
+        {
+            if (result == null)
+                return null;
+
+            Dictionary<string, object> newVersion = new Dictionary<string, object>(result);
+
+            if (!newVersion.ContainsKey("ModeNumber"))
+                newVersion.Add("ModeNumber", -1);
+
+            return newVersion;
+        }
+
+        /***************************************************/
+
+        public static Dictionary<string, object> UpgradeMeshResult(Dictionary<string, object> result)
+        {
+            if (result == null)
+                return null;
+
+            Dictionary<string, object> newVersion = new Dictionary<string, object>(result);
+
+            if (!newVersion.ContainsKey("ModeNumber"))
+                newVersion.Add("ModeNumber", -1);
+
+            //Serialisation_Engine is fallbacking to trying to de-serialise as CustomObject.
+            //When doing so it might add GUID as a property, which needs to be removed if that is the case,
+            //as MeshResult is not a BHoMObject and does not contain a BHoM_Guid.
```

Comment:
Would be nice to mention that this happens because its a collection somewhere, (not necessarily here) if that isn't a bug that's expected to be resolved.

To alleviate the next poor sod debugging.

Reply:
not because of collection, but because of sub-object that needs versioning, which is when the fallback kicks in.

Will raise issue for this to be resolved on the Engine side, for this not to be needed in the long run. Do you think that is ok, or want me to expand my already quite long comment a bit more? :)

Reply:
If it's to be resolved I'm fine with it

Reply:
expanded the comment anyway 👍 . Might as well